### ISSUE TITLE
Creating a player from pre-login containing apostrophe

### ISSUE THREAD

**Issue by BlauFeuer** [evennia/evennia#1076]
#### Brief summary of issue / Description of requested feature:
Creating a character from pre-login containing apostrophe yields unexpected and varied results, requires manual intervention.

#### Steps to reproduce the issue / Reasons for adding feature:

1. `create T'Pring Vulcan` 
2. `create "T'Pau" Vulcan`
3. `create Muad’Dib Dune`

#### Error output / Expected result of feature

The first creates a player and character named `T`

The second yields:
`Usage (without <>): create <name> <password>`
`If <name> or <password> contains spaces, enclose it in quotes.`

The third yields:
`Playername can max be 30 characters or fewer. Letters, spaces, digits and @/./+/-/_ only.`

The user who attempts this receives conflicting information and no success, requiring a manual renaming to name the character object what the user wants.

#### Extra information, such as Evennia revision/repo/branch, operating system and ideas for how to solve / implement:
Latest revision, Evennia, master branch

Escape input to allow a larger variety of character names, instruct that creating a character with a name containing an apostrophe requires special action - renaming in web admin after creation.

**Comment by robbintt** [comment]
I tested these as well, results are the same:

```
 Usage (without <>): create <name> <password>
If <name> or <password> contains spaces, enclose it in quotes.
create Muad`dib Dune

 Playername can max be 30 characters or fewer. Letters, spaces, digits and @/./+/-/_ only.
create "T'pau" Vulcan

 Usage (without <>): create <name> <password>
If <name> or <password> contains spaces, enclose it in quotes.
create T'Pring Vulcan
A new account 'T' was created. Welcome!

You can now log with the command 'connect T <your password>'.
```

**Comment by robbintt** [comment]
Submitted PR to fix this behavior by simplifying the `create` and `connect` parser.  See the bottom of https://github.com/evennia/evennia/pull/1078 for more details on future actions.

**Comment by BlauFeuer** [comment]
Since the character takes on the name of the player when both are created pre-login, this also affects character names by indirectly limiting them when created this way.

If the assumption that passwords with spaces in them will be quoted then anything left of the last quoted group is the player name  (after parsing out "connect" or "create").  I'm just concerned that there might be unseen issues with player names containing the `'` character.  I know some people who will insist on trying to use it in the name, so the instructions at pre-login need to be clear.

And if there's consequences in renaming the player after the fact, I want to be sure not to do that unless I know it's safe to do so.

**Comment by robbintt** [comment]
Added single quote support in player names and passwords: https://github.com/evennia/evennia/pull/1082/commits/e918ded8fd7a626c00aa9fc694b98f7b7d2e2cf6

Definitions: 
glyph - a single 'letter' in a language system, not necessarily a-z, includes any single mark such as !@#$%^&*() or any non european letter as well.

I can add a large number of glyphs in the glyph whitelist regular expressions, but I feel that we should determine how to add this large number of glyphs in a maintainable way.  Maybe there is a reference from Django discussing acceptable characters in usernames.

**Comment by robbintt** [comment]
Owners may also want to manage the acceptable glyphs, for example in a fantasy world you may only want to accept capital, lowercase, and single quote, and you may want to prevent duplicate accounts with the names Joe and joe if we aren't already doing that.  This requires passing a flag to the regex and impacts current users.

**Comment by Griatch** [comment]
This was closed with eb46186.
